### ISSUE TITLE
NoSuchMethodError: org.apache.commons.lang3.StringEscapeUtils.escapeXml11

### ISSUE THREAD

**Issue by timmc-bcov** [rm-hull/lein-nvd#11]
I ran `lein nvd check` in a project at work and got the following error:

```
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=512m; support was removed in 8.0
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=512m; support was removed in 8.0
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/home/timmc/.m2/repository/org/slf4j/slf4j-simple/1.7.25/slf4j-simple-1.7.25.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/home/timmc/.m2/repository/ch/qos/logback/logback-classic/1.1.2/logback-classic-1.1.2.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.SimpleLoggerFactory]
Checking dependencies for [project coordinates elided] ...
[2017-09-26 14:38:41.568] INFO Engine - Checking for updates
[2017-09-26 14:38:41.582] INFO NvdCveUpdater - Skipping NVD check since last check was within 4 hours.
[2017-09-26 14:38:41.585] INFO Engine - Check for updates complete (16 ms)
[2017-09-26 14:38:41.586] INFO Engine - Analysis Started
[2017-09-26 14:38:42.767] INFO Engine - Finished Archive Analyzer (1 seconds)
[2017-09-26 14:38:42.813] INFO Engine - Finished File Name Analyzer (0 seconds)
[2017-09-26 14:38:47.285] INFO Engine - Finished Jar Analyzer (4 seconds)
[2017-09-26 14:39:45.792] INFO Engine - Finished Central Analyzer (58 seconds)
[2017-09-26 14:39:45.834] INFO Engine - Finished Dependency Merging Analyzer (0 seconds)
[2017-09-26 14:39:45.858] INFO Engine - Finished Version Filter Analyzer (0 seconds)
[2017-09-26 14:39:45.962] INFO Engine - Finished Hint Analyzer (0 seconds)
[2017-09-26 14:39:48.871] INFO CPEAnalyzer - Created CPE Index (2 seconds)
[2017-09-26 14:39:50.930] INFO Engine - Finished CPE Analyzer (4 seconds)
[2017-09-26 14:39:50.937] INFO Engine - Finished False Positive Analyzer (0 seconds)
[2017-09-26 14:39:51.144] INFO Engine - Finished Cpe Suppression Analyzer (0 seconds)
[2017-09-26 14:39:51.483] INFO Engine - Finished NVD CVE Analyzer (0 seconds)
[2017-09-26 14:39:51.639] INFO Engine - Finished Vulnerability Suppression Analyzer (0 seconds)
[2017-09-26 14:39:51.696] INFO Engine - Finished Dependency Bundling Analyzer (0 seconds)
[2017-09-26 14:39:51.965] INFO Engine - Analysis Complete (70 seconds)
Exception in thread "main" org.apache.velocity.exception.MethodInvocationException: Invocation of method 'xml' in  class org.owasp.dependencycheck.reporting.EscapeTool threw exception java.lang.NoSuchMethodError: org.apache.commons.lang3.StringEscapeUtils.escapeXml11(Ljava/lang/String;)Ljava/lang/String; at templates/xmlReport.vsl[line 27, column 24], compiling:(/home/timmc/work/the_project/server/target/da066d96d0453d4c694f0f5a7419a2d4d8102935-init.clj:1:5)
	at clojure.lang.Compiler.load(Compiler.java:7391)
	at clojure.lang.Compiler.loadFile(Compiler.java:7317)
	at clojure.main$load_script.invokeStatic(main.clj:275)
	at clojure.main$init_opt.invokeStatic(main.clj:277)
	at clojure.main$init_opt.invoke(main.clj:277)
	at clojure.main$initialize.invokeStatic(main.clj:308)
	at clojure.main$null_opt.invokeStatic(main.clj:342)
	at clojure.main$null_opt.invoke(main.clj:339)
	at clojure.main$main.invokeStatic(main.clj:421)
	at clojure.main$main.doInvoke(main.clj:384)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at clojure.lang.Var.invoke(Var.java:383)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.Var.applyTo(Var.java:700)
	at clojure.main.main(main.java:37)
Caused by: org.apache.velocity.exception.MethodInvocationException: Invocation of method 'xml' in  class org.owasp.dependencycheck.reporting.EscapeTool threw exception java.lang.NoSuchMethodError: org.apache.commons.lang3.StringEscapeUtils.escapeXml11(Ljava/lang/String;)Ljava/lang/String; at templates/xmlReport.vsl[line 27, column 24]
	at org.apache.velocity.runtime.parser.node.ASTMethod.handleInvocationException(ASTMethod.java:261)
	at org.apache.velocity.runtime.parser.node.ASTMethod.execute(ASTMethod.java:187)
	at org.apache.velocity.runtime.parser.node.ASTReference.execute(ASTReference.java:280)
	at org.apache.velocity.runtime.parser.node.ASTReference.render(ASTReference.java:369)
	at org.apache.velocity.runtime.parser.node.ASTBlock.render(ASTBlock.java:72)
	at org.apache.velocity.runtime.directive.Foreach.render(Foreach.java:420)
	at org.apache.velocity.runtime.parser.node.ASTDirective.render(ASTDirective.java:207)
	at org.apache.velocity.runtime.parser.node.SimpleNode.render(SimpleNode.java:342)
	at org.apache.velocity.runtime.RuntimeInstance.render(RuntimeInstance.java:1378)
	at org.apache.velocity.runtime.RuntimeInstance.evaluate(RuntimeInstance.java:1314)
	at org.apache.velocity.app.VelocityEngine.evaluate(VelocityEngine.java:272)
	at org.owasp.dependencycheck.reporting.ReportGenerator.processTemplate(ReportGenerator.java:355)
	at org.owasp.dependencycheck.reporting.ReportGenerator.processTemplate(ReportGenerator.java:313)
	at org.owasp.dependencycheck.reporting.ReportGenerator.write(ReportGenerator.java:242)
	at org.owasp.dependencycheck.reporting.ReportGenerator.write(ReportGenerator.java:236)
	at org.owasp.dependencycheck.reporting.ReportGenerator.write(ReportGenerator.java:213)
	at nvd.report$generate_report.invokeStatic(report.clj:46)
	at nvd.report$generate_report.invoke(report.clj:37)
	at nvd.task.check$_main.invokeStatic(check.clj:54)
	at nvd.task.check$_main.invoke(check.clj:51)
	at user$eval317.invokeStatic(da066d96d0453d4c694f0f5a7419a2d4d8102935-init.clj:1)
	at user$eval317.invoke(da066d96d0453d4c694f0f5a7419a2d4d8102935-init.clj:1)
	at clojure.lang.Compiler.eval(Compiler.java:6927)
	at clojure.lang.Compiler.eval(Compiler.java:6917)
	at clojure.lang.Compiler.load(Compiler.java:7379)
	... 14 more
Caused by: java.lang.NoSuchMethodError: org.apache.commons.lang3.StringEscapeUtils.escapeXml11(Ljava/lang/String;)Ljava/lang/String;
	at org.owasp.dependencycheck.reporting.EscapeTool.xml(EscapeTool.java:84)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.doInvoke(UberspectImpl.java:395)
	at org.apache.velocity.util.introspection.UberspectImpl$VelMethodImpl.invoke(UberspectImpl.java:384)
	at org.apache.velocity.runtime.parser.node.ASTMethod.execute(ASTMethod.java:173)
	... 37 more
Subprocess failed
```

Leiningen version 2.7.1 on Java 1.8.0_31. I can try pruning this project down to make an SSCCE, but wanted to go ahead and file the bug (especially in case it's a known one.)

**Comment by rm-hull** [comment]
Not seen this before;  It looks like there is _possibly_ some classpath conflict with commons-lang / common-lang3 - this might be down to some misconfig inside DependencyCheck:

```
 [org.owasp/dependency-check-core "2.1.1"]
   [com.google.code.gson/gson "2.4"]
   [com.h2database/h2 "1.4.196" :scope "runtime"]
   [com.sun.mail/mailapi "1.5.6"]
   [commons-io "2.5"]
   [joda-time "1.6"]
   [org.apache.commons/commons-compress "1.14"]
   [org.apache.commons/commons-lang3 "3.4"]            <=====
   [org.apache.lucene/lucene-analyzers-common "4.7.2"]
   [org.apache.lucene/lucene-core "4.7.2"]
   [org.apache.lucene/lucene-queryparser "4.7.2"]
     [org.apache.lucene/lucene-queries "4.7.2" :exclusions [[org.apache.lucene/lucene-core]]]
     [org.apache.lucene/lucene-sandbox "4.7.2" :exclusions [[org.apache.lucene/lucene-core] [jakarta-regexp]]]
   [org.apache.velocity/velocity "1.7"]
     [commons-collections "3.2.1"]
     [commons-lang "2.4"]                              <=====
   [org.glassfish/javax.json "1.0.4"]
   [org.jsoup/jsoup "1.10.2"]
   [org.owasp/dependency-check-utils "2.1.1"]
``` 

Just out of interest, if you run `lein deps :tree | grep commons-lang` in your project, does it show a version?

**Comment by rm-hull** [comment]
Not been able to reproduce the error on a few different projects here.
Could you try downgrading to `[lein-nvd "0.2.7"]` and see if crashes with that version?

**Comment by timmc-bcov** [comment]
Yes, I see `[org.apache.commons/commons-lang3 "3.1"]` and `[commons-lang "2.5"]` in the deps.

Still happens on lein-nvd 0.2.7.

**Comment by timmc-bcov** [comment]
Adding `[org.apache.commons/commons-lang3 "3.1"]` to a project causes the issue with `[lein-nvd 0.3.0]`.

**Comment by rm-hull** [comment]
I managed to reproduce the same crash with: 
* `[org.apache.commons/commons-lang3 "3.0"]`
* `[org.apache.commons/commons-lang3 "3.1"]`
* `[org.apache.commons/commons-lang3 "3.2"]`

The following versions all appeared to work OK however:
* `[org.apache.commons/commons-lang3 "3.3"]`
* `[org.apache.commons/commons-lang3 "3.4"]`
* `[org.apache.commons/commons-lang3 "3.5"]`
* `[org.apache.commons/commons-lang3 "3.6"]`

**Comment by vemv** [comment]
I actually saw this precise error (`Invocation of method 'xml' in  class org.owasp.dependencycheck.reporting...`) while hacking on lein-nvd.

The outcome was a series of internal improvements and the following guide: https://github.com/rm-hull/lein-nvd#avoiding-classpath-interference

I believe that should suffice to consider this issue fixed (as of 1.5.0), would love to know if it worked for you.

Cheers - V
